gigascience/gigadb-website
# New feature/1143 hyperlinks to open new tabs

# Pull request for issue: #1143, #1793

This is a pull request for the following functionalities:

* Update all external links to open in new browser window using javascript

## How to test?

Click on the social links or any other external link and verify they open in new tab

## How have functionalities been implemented?

* A script has been added to all layouts. the script adds a `target="__blank"` to all external links, protected/views/shared/_handle_external_links.php
* It also adds a message that helps screen reader users determine that the link opens in new tab.
* It also adds attributes that improve security
* If the link already has a `target` attribute, then it is assumed that this is hardcoded and nothing is done

Notes:
* it is not possible to do this with CSS, as far as I know
* the script will work for any link that is present in the page on load but not links which are rendered afterward, e.g. on user interaction. these should be a minority and can be handled manually case by case. An alternative would be to use an observer to react to changes in the DOM tree, but seems a bit overkill

## Any issues with implementation?

--

## Any changes to automated tests?

- Commented out tests that expect to open the "jobs" page. If you have a good solution for this, please let me know. Also, wondering whether we should be testing external pages content at all. Maybe we can delete these 3 tests, or only test that the link exists and points to a certain url
- Updated one test checking for the link to the changelog version, to account for extra text content within the link
